### ISSUE TITLE
NullPointerException in DeviceInfo.getCountry()

### ISSUE THREAD

**Issue by benjamin-bader** [amplitude/Amplitude-Android#44]
Observed in v1.6.3 on a Samsung Galaxy Grand Prime (SM-G530FZ), Android 4.4.4.

```
java.lang.NullPointerException
       at android.telephony.MultiSimTelephonyManager.isVoiceCapable(MultiSimTelephonyManager.java:441)
       at android.telephony.TelephonyManager.getPhoneType(TelephonyManager.java:571)
       at com.amplitude.api.DeviceInfo.getCountryFromNetwork(DeviceInfo.java:112)
       at com.amplitude.api.DeviceInfo.getCountryUncached(DeviceInfo.java:135)
       at com.amplitude.api.DeviceInfo.getCountry(DeviceInfo.java:76)
       at com.amplitude.api.AmplitudeClient$1.run(AmplitudeClient.java:137)
       at android.os.Handler.handleCallback(Handler.java:733)
       at android.os.Handler.dispatchMessage(Handler.java:95)
       at android.os.Looper.loop(Looper.java:136)
       at android.os.HandlerThread.run(HandlerThread.java:61)
```

**Comment by curtisliu** [comment]
Are there other devices that you are seeing this issue on? It looks like a NPE coming from the Android builtins. We can catch it, but strange that it's happening to begin with.

**Comment by benjamin-bader** [comment]
I agree, it is quite strange.  So far this is the only device that has this crash.

**Comment by ajalt** [comment]
This was observed again in a Samsung Galaxy Ace 4 (SM-G357FZ), Android 4.4.4.

    java.lang.NullPointerException
           at android.telephony.TelephonyManager.isVoiceCapable(TelephonyManager.java:1835)
           at android.telephony.TelephonyManager.getPhoneType(TelephonyManager.java:569)
           at com.amplitude.api.DeviceInfo.getCountryFromNetwork(DeviceInfo.java:112)
           at com.amplitude.api.DeviceInfo.getCountryUncached(DeviceInfo.java:135)
           at com.amplitude.api.DeviceInfo.getCountry(DeviceInfo.java:76)
           at com.amplitude.api.AmplitudeClient$1.run(AmplitudeClient.java:137)
           at android.os.Handler.handleCallback(Handler.java:733)
           at android.os.Handler.dispatchMessage(Handler.java:95)
           at android.os.Looper.loop(Looper.java:136)
           at android.os.HandlerThread.run(HandlerThread.java:61)

**Comment by munix** [comment]
This was observed in Lenovo A680, Android 4.2.2

http://crashes.to/s/c6aef9245ec

java.lang.NullPointerException
       at android.location.Address$1.createFromParcel(Address.java:480)
       at android.location.Address$1.createFromParcel(Address.java:476)
       at android.os.Parcel.readTypedList(Parcel.java:1761)
       at android.location.ILocationManager$Stub$Proxy.getFromLocation(ILocationManager.java:758)
       at android.location.Geocoder.getFromLocation(Geocoder.java:133)
       at com.amplitude.api.DeviceInfo.getCountryFromLocation(DeviceInfo.java:93)
       at com.amplitude.api.DeviceInfo.getCountryUncached(DeviceInfo.java:130)
       at com.amplitude.api.DeviceInfo.getCountry(DeviceInfo.java:76)
       at com.amplitude.api.AmplitudeClient$1.run(AmplitudeClient.java:143)
       at android.os.Handler.handleCallback(Handler.java:800)
       at android.os.Handler.dispatchMessage(Handler.java:100)
       at android.os.Looper.loop(Looper.java:194)
       at android.os.HandlerThread.run(HandlerThread.java:60)

**Comment by benjamin-bader** [comment]
Hi, any update on this issue?  We continue to see this and similar crashes in `#getCountryFromLocation`.

Since it's pretty clear that this is an unreliable system API, it would be helpful to catch the error and not crash.

```
android.content.res.Resources$NotFoundException: Resource ID #0x111003f
       at android.content.res.Resources.getValue(Resources.java:2354)
       at android.content.res.Resources.getBoolean(Resources.java:2110)
       at android.telephony.TelephonyManager.isVoiceCapable(TelephonyManager.java:1768)
       at android.telephony.TelephonyManager.getPhoneType(TelephonyManager.java:519)
       at com.amplitude.api.DeviceInfo.getCountryFromNetwork(DeviceInfo.java:112)
       at com.amplitude.api.DeviceInfo.getCountryUncached(DeviceInfo.java:135)
       at com.amplitude.api.DeviceInfo.getCountry(DeviceInfo.java:76)
       at com.amplitude.api.AmplitudeClient$1.run(AmplitudeClient.java:143)
       at android.os.Handler.handleCallback(Handler.java:733)
       at android.os.Handler.dispatchMessage(Handler.java:95)
       at android.os.Looper.loop(Looper.java:146)
       at android.os.HandlerThread.run(HandlerThread.java:61)
```

**Comment by djih** [comment]
Hi, we are working on this now. Will push a new release once we get the fix in.

**Comment by djih** [comment]
This is fixed with v2.0.1
